### ISSUE TITLE
Questions of the inference strategy

### ISSUE THREAD

**Issue by SchweitzerGAO** [OpenGVLab/SDLM#5]
Hello, I am wondering why you keep the first token of each block clean during the training phase, which leads to the necessity of duplicating the last decoded token during inference. Is this mandatory?
<img width="756" height="741" alt="Image" src="https://github.com/user-attachments/assets/4ef3c6bb-7697-43f8-ac9e-b1b8ef97a486" />
Besides, does the image below correctly demonstrate the attention mask during inference? If so, is the attention to x3 and x3' redundant?

<img width="741" height="713" alt="Image" src="https://github.com/user-attachments/assets/5e7af09d-2668-4817-91c6-eb72f1c7ffc8" />

**Comment by lll2343** [comment]
Thank you for your question. That's an excellent point, and my apologies for not including this explanation in the paper. To clarify, during inference, the attention mask is structured as follows:

<img width="449" height="727" alt="Image" src="https://github.com/user-attachments/assets/0f764969-d4f0-4959-ba0d-59484a2baf77" />

* For instance, as shown in the mask you provided, the first token $x_3$  is non-visible to the masked block.

The corresponding code is located here. For details, you can refer to: https://github.com/OpenGVLab/SDLM/blob/main/sdlm/model/sdlm_qwen2_5/attn_mask_utils.py#L178

**Comment by SchweitzerGAO** [comment]
Thanks!
